Explorer/Assets/DCL/Multiplayer/Connections/Archipelago/Rooms/ForkArchipelagoIslandRoom.cs

```diff
@@ -0,0 +1,58 @@
+using Cysharp.Threading.Tasks;
+using DCL.Multiplayer.Connections.Archipelago.AdapterAddress.Current;
+using DCL.Multiplayer.Connections.Rooms;
+using DCL.Multiplayer.Connections.Rooms.Connective;
+using LiveKit.Rooms;
+using System;
+using System.Threading;
+
+namespace DCL.Multiplayer.Connections.Archipelago.Rooms
+{
+    public class ForkArchipelagoIslandRoom : IArchipelagoIslandRoom
```

Comment:
Why we should keep both `ForkArchipelagoIslandRoom` and `ArchipelagoIslandRoom`. In our domain are they interchangeable? Or there is no option but to use the fork one?

Reply:
Do you suggest to merge them? It would look pretty ugly, because each of it contains a big amount of behaviour (also counting FixedAdapterConnectiveRoom) Fork composes them

Reply:
For me it's confusing that the logic of `ForkArchipelagoIslandRoom` acts as a factory but it's a decorator instead.
Why it can't be a factory then? Then it will be clear that we are using factory instead of `IArchipelagoIslandRoom`.
There should be some common sense limitations for using `Decorator` pattern. 
I truly believe that on the consumer side it's more predictable if the factory is used directly and not something abstract hidden behind several layers of nesting.

Reply:
The same concern related to `RenewableConnectiveRoom`. Why it should implement `IConnectiveRoom` interface when it acts as a factory.
Why do we place it on the same layer as a decorator?